src/tms_service.ts

```diff
@@ -98,11 +76,11 @@ export class TMSService extends AbstractEmsService {
       const { sources } = vectorJson;
       for (const sourceName of Object.getOwnPropertyNames(sources)) {
         const { url } = sources[sourceName];
-        const sourceUrl = this._proxyPath + this._getAbsoluteUrl(url);
+        const sourceUrl = this._proxyPath + this._getAbsoluteUrl(url!);
         const extendedUrl = this._emsClient.extendUrlWithParams(sourceUrl);
-        const sourceJson = await this._emsClient.getManifest<EmsVectorSource>(extendedUrl);
+        const sourceJson = await this._emsClient.getManifest<MBVectorSource>(extendedUrl);
 
-        const extendedTileUrls = sourceJson.tiles.map((tileUrl: string) => {
+        const extendedTileUrls = sourceJson.tiles!.map((tileUrl: string) => {
```

Comment:
Instead of `!`, how about checking exists first? VectorSource does have tiles typed as "string[] | undefined" so tiles could be undefined.